Sources/Datadog/RUM/Instrumentation/Actions/SwiftUI/SwiftUIActionModifier.swift

```diff
@@ -0,0 +1,103 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+#if canImport(SwiftUI)
+import SwiftUI
+
+/// `SwiftUI.ViewModifier` for RUM which invoke `addUserAction` from the
+/// global RUM Monitor when the modified view receives a tap.
+@available(iOS 13, *)
+internal struct RUMTapActionModifier: SwiftUI.ViewModifier {
+    /// The required number of taps to complete the tap action.
+    let count: Int
+
+    /// Action Name used for RUM Explorer.
+    let name: String
+
+    /// Custom attributes to attach to the Action.
+    let attributes: [AttributeKey: AttributeValue]
+
+    func body(content: Content) -> some View {
+        content.simultaneousGesture(
```

Comment:
`SwiftUI` newbie question - it is not possible to use generics / protocol constraints to know that `Content` is for example `NavigationLink` and read its `Text` then `label` to have the `name` inferred automatically (e.g. `"Push to Next View"`)?

Reply:
Indeed, I would have liked that very much... Definition of `NavigationLink` or `Button`, or any other components I encountered does not expose anything of their state or content. I also tried to use accessibility info but I couldn't infer anything from the `ViewModifier.Content` or the `SwiftUI.View` itself.
I might be missing something tho, but I haven't been able to infer any valuable information from the view so far.